src/main/java/com/coveros/selenified/element/Get.java

```diff
@@ -369,83 +368,34 @@ public int numOfTableRows() {
-        }
-        if (!element.is().table()) {
-            return null;    // returning an empty array could be confused with no columns
-        }
-        List<WebElement> rows = tableRows();
-        List<WebElement> row = tableRow(1);
-        List<List<WebElement>> columns = new ArrayList<>();
-        for (int i = 0; i < row.size(); i++) {
-            List<WebElement> column = new ArrayList<>();
-            for (int j = 0; j < rows.size(); j++) {
-                List<WebElement> cells = tableRow(j);
-                column.add(cells.get(i));
-            }
-            columns.add(column);
-        }
-        return columns;
+        Element thCells = rows.findChild(app.newElement(Locator.TAGNAME, "th"));
+        Element tdCells = rows.findChild(app.newElement(Locator.TAGNAME, "td"));
+        return thCells.get().matchCount() + tdCells.get().matchCount();
     }
 
     /**
      * Retrieves a specific row from the element. If the element isn't present
-     * or a table, a null value will be returned. If the specified row is out of
-     * range, an empty list is returned
+     * or a table, or doesn't have that many rows a null value will be returned.
      *
      * @param rowNum - the row number of the table to obtain - note, row numbering
-     *               starts at 1, NOT 0
+     *               starts at 0
      * @return List: a list of the table cells in the row as WebElements
      */
     @SuppressWarnings("squid:S1168")
-    public List<WebElement> tableRow(int rowNum) {
-        List<WebElement> rows = tableRows();
+    public Element tableRow(int rowNum) {
+        Element rows = tableRows();
         if (rows == null) {
-            return null;    // returning an empty array could be confused with a row out of range
-        }
-        if (rows.size() < rowNum) {
-            return new ArrayList<>();
-        }
-        WebElement thisRow = rows.get(rowNum);
-        List<WebElement> cells = thisRow.findElements(By.xpath(".//th|.//td"));
-        return new ArrayList<>(cells);
-    }
-
-    /**
-     * Retrieves a specific column from the element. If the element isn't
-     * present or a table, a null value will be returned. If the specified row
-     * is out of range, an empty list is returned
-     *
-     * @param colNum - the column number of the table to obtain - note, column
-     *               numbering starts at 1, NOT 0
-     * @return List: a list of the table cells in the column as WebElements
-     */
-    @SuppressWarnings("squid:S1168")
-    public List<WebElement> tableColumn(int colNum) {
-        List<List<WebElement>> columns = tableColumns();
-        if (columns == null) {
-            return null;    // returning an empty array could be confused with a column out of range
+            return null;
         }
-        if (columns.size() < colNum) {
-            return new ArrayList<>();
+        int i = numOfTableRows();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "i". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "i" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1481)